lib/node/workspace.dart

```diff
@@ -25,13 +25,193 @@ class ViewRegistry extends ProxyHolder {
     _instance = this;
   }
 
-  // TODO: expose addViewProvider(providerSpec)
+  // TODO: add modelConstructor
+  Disposable addViewProvider(createView) =>
+      invoke('addViewProvider', createView);
 
   /// Get the view associated with an object in the workspace. The result is
   /// likely an html Element.
   dynamic getView(object) => invoke('getView', object);
 }
 
+// A Change Listener for events going into Atom instead of out.
+class JsChangeListener {
+  List<JsFunction> _callbacks = [];
+
+  void add(JsFunction callback) {
+    _callbacks.add(callback);
+    return jsify ({
+      'dispose': () => _callbacks.remove(callback)
+    });
+  }
+
+  void change(List parameters) {
+    for (var callback in _callbacks) {
+      callback?.apply(parameters);
+    }
+  }
+}
+
+/// The term "item" refers to anything that can be displayed
+/// in a pane within the workspace, either in the {WorkspaceCenter} or in one
+/// of the three {Dock}s. The workspace expects items to conform to the
+/// following interface:
+class Item extends ProxyHolder {
+
+  JsChangeListener onDidChangeTitle = new JsChangeListener();
+
+  String _title;
+
+  Item(JsObject object) : super(object);
+
+  Item.fromFields({element, title, uri, defaultLocation, destroy})
+      : _title = title,
+        super(jsify({
+    'element': element,
+    // Returns the URI associated with the item.
+    'getURI': () => uri,
+    // Tells the workspace where your item should be opened in absence of a user
+    // override. Items can appear in the center or in a dock on the left, right, or
+    // bottom of the workspace.
+    //
+    // Returns a {String} with one of the following values: `'center'`, `'left'`,
+    // `'right'`, `'bottom'`. If this method is not defined, `'center'` is the
+    // default.
+    'getDefaultLocation': () => defaultLocation,
+    // Destroys the item. This will be called when the item is removed from its
+    // parent pane.
+    'destroy': destroy,
+  })) {
+    // Returns a {String} containing the title of the item to display on its
+    // associated tab.
+    obj['getTitle'] = jsify(() => _title);
+    // Called by the workspace so it can be notified when the item's title changes.
+    // Must return a {Disposable}.
+    obj['onDidChangeTitle'] = jsify(onDidChangeTitle.add);
+
+    // Custom functions.  Used to reach original Item objects, from
+    // passed in Item references.  I.e., go through item functions instead
+    // of using local private fields that maybe not be an Item constructed
+    // with JsObject ctor.
+    obj['setTitle'] = jsify((String newTitle) {
+      _title = newTitle;
+      onDidChangeTitle.change([newTitle]);
+    });
+  }
+
+  // Any function below must go through invoking a JsFunction, because
+  // private fields might not be in this instance of Item.
```

Comment:
nit: you may want a newline between this comment and the getter, so it looks less like dartdoc